File: remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        if len(nums) <= 0:
            return 0
        
        prev_idx = -1
        for idx, num in enumerate(nums):
            if num != val:
                prev_idx += 1
                nums[prev_idx] = num
                
        return prev_idx + 1

File: remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums.length <= 0) {
            return 0;
        }
        int prev_idx = -1;
        for(int i = 0; i < nums.length; i++) {
            if(val != nums[i]) {
                nums[++prev_idx] = nums[i];
            }
        }
        return prev_idx + 1;
    }
}